core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -934,29 +958,55 @@ public Void visitTableScan(TableScanNode node, Void context)
         public Void visitValues(ValuesNode node, Void context)
         {
             NodeRepresentation nodeOutput = addNode(node, "Values");
-            if (node.getRows().isEmpty()) {
-                for (int i = 0; i < node.getRowCount(); i++) {
-                    nodeOutput.appendDetailsLine("()");
-                }
+            if (node.getRows().isEmpty() || node.getRows().get().isEmpty()) {
+                nodeOutput.appendDetailsLine(format("(), row count: %s", node.getRowCount()));
```

Comment:
I think it would be more readable if output like:
```
()
()
()
()
```
followed the same truncation threshold.